version: "3.8"

services:
  postgresql:
    image: postgres:16.1-alpine3.18
    container_name: postgresql
    command:
      - "postgres"
      - "-c"
      - "config_file=/etc/postgresql.conf"
    environment:
      POSTGRES_DB: "tg_bot"
      POSTGRES_USER: "admin"
      POSTGRES_PASSWORD: "adminpassword"
      PGDATA: "/var/lib/postgresql/data/pgdata"
    volumes:
      - ./postgresql/Init_Database:/docker-entrypoint-initdb.d
      - ./postgresql/data:/var/lib/postgresql/data
      - ./postgresql/postgresql.conf:/etc/postgresql.conf:ro
    ports:
      - "5432:5432"
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U admin -d tg_bot" ]
      interval: 30s
      timeout: 5s
      retries: 5
      start_period: 10s
    restart: unless-stopped
    deploy:
      resources:
        limits:
          cpus: '1'
          memory: 4G
    networks:
      - mynet


  tg_bot:
    container_name: tg_bot
    build: ./
    restart:
      always
    networks:
      - mynet
    depends_on:
      postgresql:
        condition: service_healthy



networks:
  mynet:
